docs/DevEnvironment.md

```diff
@@ -0,0 +1,158 @@
+`rouge/`.
+
+## Installing Development Dependencies
+
+Ruby provides support for using external "packages" of Ruby code. These packages
+are called _gems_. While Rouge itself does not depend on any gems to perform
+syntax highlighting, it does have a number of dependencies that you need to
+install to _develop_ Rouge. These are the development dependencies.
+
+### Installing Bundler
+
+The easiest way to install Rouge's development dependencies is using
+[Bundler][]. Bundler is itself a gem and we'll install it differently to how we
+install the other dependencies.
+
+[Bundler]: https://bundler.io/
+
+If you already develop with Ruby, you no doubt have Bundler installed. You can
+check if you do by typing `bundle -v` at the command line. If you don't see the
+version number then you need to install Bundler. To do this, type:
+
+```shell
+gem install bundler
+```
+
+Ruby's `gem` tool will grab the latest Bundler package and install it. Once
+that's complete, you're ready to rock.
+
+### Installing the Dependencies
+
+Rouge comes with a list of gems it depends upon called a _Gemfile_. Make sure
+you're at the top level of your clone of your repository and type:
+
+```shell
+bundle install --path vendor
+```
+
+This command tells Bundler to install the gems Rouge in the Gemfile into a
+directory called `vendor/`. This has one drawback (explained below) but means
+the gems we use for Rouge are isolated from the other gems we may have installed
+on our system. This will be tremendously helpful in avoiding conflicts that
+arise because of the use of incompatible versions of a gem.
+
+The one drawback is that we will need to tell Ruby every time we run our Rouge
+code that it needs to look for the gems in `vendor/`. Bundler makes this easy by
+providing the command `bundle exec`. So if we want to run our Rake tests, we
+type `bundle exec rake` rather than just `rake`.
+
+## Using Branches
+
+It's best to develop in a _branch_. You can create a branch by typing:
+
+```shell
+git checkout -b <name_of_your_branch>
+```
+
+You don't need to do it this way but Rouge maintainers often use the format 
+`feature.<name>` for features and `bugfix.<name>` for bug fixes.
+
+Using branches will make it easier for others to collaborate with you and make
+it easier for you to start fresh if you screw things up from your last stable
+state. You can read more about branches on [GitHub][gh-branch-docs].
+
+[gh-branch-docs]: https://help.github.com/en/articles/about-branches
+
+## Next Steps
+
+You're now ready to roll. Here are the things you can do from the top level of
+your cloned repository:
+
+1. **Run the Visual Test App**
+   Rouge includes a little web app you can run to display highlighted code. You
+   can run this by typing `bundle exec rackup`. By default, this will start a
+   web server on port 9292. You can access it by going to
+   <http://localhost:9292> with Rack running. If everything is working, you'll
```

Comment:
So that it's reads better on GitHub and can be easily copied..
```suggestion
   `http://localhost:9292` with Rack running. If everything is working, you'll
```

Reply:
Not sure what you mean. The link works fine on GitHub when surrounded by arrow tags.